src/main/java/net/dries007/tfc/common/blocks/devices/ChannelBlock.java

```diff
@@ -0,0 +1,203 @@
+            .filter(facing -> facing.getKey() != Direction.UP).collect(Util.toMap());
+
+    public static final BooleanProperty NORTH = PipeBlock.NORTH;
+    public static final BooleanProperty EAST = PipeBlock.EAST;
+    public static final BooleanProperty SOUTH = PipeBlock.SOUTH;
+    public static final BooleanProperty WEST = PipeBlock.WEST;
+    public static final BooleanProperty DOWN = PipeBlock.DOWN;
+
+    public static final BooleanProperty TRIGGERED = BlockStateProperties.TRIGGERED;
+    public static final BooleanProperty WITH_METAL = BooleanProperty.create("with_metal");
+
+    private static final VoxelShape[] SHAPES = new VoxelShape[16];
+
+    static {
+        final VoxelShape base = box(5.0D, 0.0D, 5.0D, 11.0D, 5.0D, 11.0D);
+
+        final VoxelShape east = box(11.0D, 0.0D, 5.0D, 16.0D, 5.0D, 11.0D);
+        final VoxelShape south = box(5.0D, 0.0D, 11.0D, 11.0D, 5.0D, 16.0D);
+        final VoxelShape west = box(0.0D, 0.0D, 5.0D, 5.0D, 5.0D, 11.0D);
+        final VoxelShape north = box(5.0D, 0.0D, 0.0D, 11.0D, 5.0D, 10.0D);
+
+        final VoxelShape[] directions = new VoxelShape[] { south, west, north, east };
+
+        for (int i = 0; i < SHAPES.length; i++) {
+            VoxelShape shape = base;
+            for (Direction direction : Direction.Plane.HORIZONTAL) {
+                if (((i >> direction.get2DDataValue()) & 1) == 1) {
+                    shape = Shapes.or(shape, directions[direction.get2DDataValue()]);
+                }
+            }
+            SHAPES[i] = shape;
+        }
+    }
+
+    private static BlockState updateConnectedSides(LevelAccessor level, BlockPos pos, BlockState state) {
+        for (Direction dir : Direction.values()) {
+            if (dir == Direction.UP)
+                continue;
+
+            // When going down, allow >1 block distance
+            byte maxDistance = dir == Direction.DOWN ? Byte.MAX_VALUE : 1;
+
+            boolean isAdjacentConnectable = false;
+
+            for (byte i = 1; i < maxDistance + 1; i++) {
+                BlockPos relative = pos.relative(dir, i);
+                BlockState blockState = level.getBlockState(relative);
+                Block block = blockState.getBlock();
+
+                if (block instanceof ChannelBlock || block instanceof CrucibleBlock || block instanceof MoldBlock) {
+                    isAdjacentConnectable = true;
+                    break;
+                } else if (!blockState.isAir()) {
+                    break;
+                }
+            }
+
+            state = state.setValue(DirectionPropertyBlock.getProperty(dir), isAdjacentConnectable);
+        }
+
+        return state;
+    }
+
+    @Override
+    public boolean onDestroyedByPlayer(BlockState state, Level level, BlockPos pos, Player player, boolean willHarvest,
+            FluidState fluid) {
+        // On destroy, notify source channel that all flows going through this
+        // channel have been broken
+        if (!level.isClientSide()) {
```

Comment:
Is there a reason this is only checked on the server? Sometimes we prefer to run these things on both sides to prevent desync